source/common/http/header_map_impl.cc

```diff
@@ -464,6 +464,17 @@ void HeaderMapImpl::remove(const LowerCaseString& key) {
   }
 }
 
+void HeaderMapImpl::remove(const std::regex& regex) {
+  for (auto i = headers_.begin(); i != headers_.end();) {
+    absl::string_view key = i->key().getStringView();
+    if (std::regex_search(key.begin(), key.end(), regex)) {
+      i = headers_.erase(i);
```

Comment:
I think this is not going to work correctly if this is a match of an inline header. You are probably going to need to do some extra work to make it work... Can you add a test case?

Reply:
And my intuition of "I should wait an hour and see what Matt thinks" pans out once again :-D